pkg/config/configuration.go

```diff
@@ -451,7 +451,7 @@ func getSpiffeID(ctx context.Context) (string, error) {
 
 // IsOperationAllowedByAccessControlPolicy determines if access control policies allow the operation on the target app
 func IsOperationAllowedByAccessControlPolicy(spiffeID *SpiffeID, srcAppID string, inputOperation string, httpVerb common.HTTPExtension_Verb, appProtocol string, accessControlList *AccessControlList) (bool, string) {
-	if accessControlList == nil {
+	if accessControlList == nil || spiffeID == nil {
```

Comment:
This will create a security hole. If there's an access control list in place, any malicious code will be able to bypass the check by not having the SPIFFE ID.

Reply:
The correct behavior is to perform the check (and require the existence of a Spiffe ID) if there's an access control list in place AND if mTLS is enabled. (which is a prerequisite for Access Control).

Reply:
As I commented [here](https://github.com/dapr/dapr/issues/2114#issuecomment-699092901), in the self-hosted quickstart, there was no Configuration, means means the `accessControlList` should have been nil.

Reply:
I agree....realized that when the UT failed. Will change the fix

Reply:
fixed